### PR TITLE
Restore backup assemblies when sigkill

### PR DESCRIPTION
**Pull request by ViktorHofer**
With sigkill or other execution interruptions the instrumented assemblies will remain in the output directory and the file backups won't be restored. Adding a cache to restore the backups in such cases.

cc @MarcoRossignoli @tonerdo

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
@ViktorHofer let me understand better the case...if the "test process" is "interrupted" for some reason why recover?I mean if it's stopped will be eventually replaced by new build/test, no?

**Comment by ViktorHofer** [comment]
If you want to "safely" instrument assemblies - means the output should be exactly the same as before - a sigkill could cause big troubles here. We are hitting this in corefx whenever a coverage measurement takes too long and the user kills the process. We don't rebuild all the assemblies that we use during the measurement, i.e. CoreLib.

But this isn't tied to corefx at all. Since the `--include-directory` option was added, additional assembly directories can be specified. Usually those additional inputs aren't rebuilt and should not get into a dirty state.

File: src/coverlet.core/Helpers/InstrumentationHelper.cs [hunk 1]
```diff
@@ -14,6 +14,13 @@ namespace Coverlet.Core.Helpers
 {
     internal static class InstrumentationHelper
     {
+        private static readonly Dictionary<string, string> _backupList = new Dictionary<string, string>();
```
Comment (MarcoRossignoli) [inline src/coverlet.core/Helpers/InstrumentationHelper.cs]: sooner or later we should remove al these statics and use msbuild input/output https://github.com/tonerdo/coverlet/issues/364

**Comment by MarcoRossignoli** [comment]
@tonerdo CI fail with some test on cecil tests...is it never happened before?No issue on my local.

**Comment by tonerdo** [comment]
@MarcoRossignoli taking a look now

**Comment by ViktorHofer** [comment]
Is there ready to be merged?

**Comment by MarcoRossignoli** [comment]
@ViktorHofer in future I'll remove that static https://github.com/tonerdo/coverlet/issues/387 for now if it's a blocking issue LGTM cc: @tonerdo

**Comment by ViktorHofer** [comment]
Yeah let's merge this first and ideally publish a new package first and afterwards do the refactoring.

**Pull request merged**